express-notes-api/index.js

```diff
@@ -0,0 +1,85 @@
+const express = require('express');
+const app = express();
+const PORT = 3000;
+
+let nextId = 1;
+const notes = {};
+
+const JSONparser = express.json();
+app.use(JSONparser);
+
+app.get('/api/notes/:id', (req, res) => {
+  if (+req.params.id !== ~~req.params.id || +req.params.id <= 0) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  } else if (notes[req.params.id]) {
+    res.status(200).send(notes[req.params.id]);
+  } else {
+    res.status(404).send({ error: `cannot find note with id ${req.params.id}` });
+  }
+});
+
+app.get('/api/notes/', (req, res) => {
+  res.send(noteArray(notes));
+});
+
+app.post('/api/notes', (req, res) => {
+  if (!Object.keys(req.body).includes('content')) {
+    res.status(400).send({ error: 'content is a required field' });
+  } else {
+    try {
+      notes[nextId] = {
+        content: req.body.content,
+        id: nextId
+      };
+      res.status(201).send(notes[nextId]);
+      nextId++;
+    } catch (error) {
+      console.error(error);
+      res.status(500).send({ error: 'an unexpected error occured' });
+    }
```

Comment:
After adding a new note to the `data` you need to save the changes to the `data.json` file using `fs.writeFile` once the write operation is complete if there is an error you should send a 500 response, else if no error send the 201 response.